packages/@aws-cdk/core/lib/secret-value.ts

```diff
@@ -96,6 +110,20 @@ export class SecretValue extends Intrinsic {
 
     return new SecretValue(param.value);
   }
+
+  /**
+   * Specifies the secretArn when it was provided for a value from SecretsManager
+   */
+  readonly secretArn?: string;
+
+  constructor(value: any, options: IntrinsicProps = {}, secretArn?: string) {
+    super(value, options);
+
+    Object.defineProperty(this, SECRET_VALUE_SYMBOL, { value: true });
+    if (secretArn) {
+      this.secretArn = secretArn;
+    }
+  }
```

Comment:
If we want a new constructor argument for this class, we need a new props interface instead:

```ts
export interface SecretValueProps extends IntrinsicProps {
  readonly secretQualifier?: SecretQualifier;
}
```

And then:

```suggestion
  constructor(value: any, props: SecretValueProps = {}) {
    super(value, props);
    this.secretQualifier = props.secretQualifier;
  }
```